lib/Form/DisplayFormInWPContent.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace MailPoet\Form;
+
+use MailPoet\Entities\FormEntity;
+use MailPoet\WP\Functions as WPFunctions;
+
+class DisplayFormInWPContent {
+
+  /** @var WPFunctions */
+  private $wp;
+
+  /** @var FormsRepository */
+  private $formsRepository;
+
+  public function __construct(WPFunctions $wp, FormsRepository $formsRepository) {
+    $this->wp = $wp;
+    $this->formsRepository = $formsRepository;
+  }
+
+  // TODO remove transient in the api on form save
+
+  public function display(string $content): string {
+    $result = $content;
+    if (!$this->wp->isSingle()) return $result;
```

Comment:
I can see we have a test for this but for a simple comment saying that this code ensures that we display the form only on a page which is related to single post would be nice here.